### DISCUSSION TITLE
Exposición «Cifrado por bloques»: Preguntas y respuestas

### DISCUSSION THREAD

**Discussion by gwolf** [unamfi/sistop-2022-1#193] [Q&A]
Abro este foro como un espacio de discusión para la exposición que prepararon los alumnos Esteban Tafolla y Erick Vázquez, «Cifrado por bloques». El video está disponible en:

https://youtu.be/jTi5ZjwXqy0

Pueden consultar también el documento de texto en que los alumnos presentan este mismo tema:

https://github.com/unamfi/sistop-2022-1/blob/main/exposiciones/TafollaEsteban-VazquezErick/Investigacion_Cifrado_por_bloques.pdf

Además de participar aquí, les pido que llenen la pequña encuesta que está disponible en:

https://www.questionpro.com/t/ARiwuZqIss

**Comment by Inakyord** [comment]
Hola, espero estén bien. La criptografía y los cifrados siempre me han parecido interesantes, por lo que me gustó el tema. Creo que la duda más grande que me quedó fue ¿Cómo se puede hacer un intercambio seguro de la llave? Mencionan que es la parte más vulnerable del proceso. Otra cosa que me llamó la atención fue que sobre todo para AES la encriptación se hace mediante operaciones a nivel de bits y no del contenido (para humanos), ¿esto es porque es más eficiente o más difícil de descifrar u otra razón? Gracias.

**Reply by Erickescorpions** [reply]
Hola compañero, muchas gracias igual esperamos te encuentres bien. 

Contestando tu primer pregunta para intercambiar la llave hoy en día se hace uso de una mezcla de diferentes técnicas, por ejemplo se realiza el cifrado mediante un algoritmo simétrico como Triple DES o AES y mediante un algoritmo de cifrado asimétrico se genera una sola llave, el algoritmo que conozco se llama "**Intercambio de claves Diffie-Hellman**" que consiste en que ambos sistemas generan dos números, uno en común mediante un canal inseguro y uno se mantiene en secreto dentro de cada sistema. Con estos dos números se realiza una operación modular y se vuelve a intercambiar el resultado mediante un canal inseguro, obvio nunca se expone esos números secretos de cada sistema pero eligiendo bien los números a usar (que generalmente son muy grandes y se utilizan números primos) es muy difícil que un atacante pueda descubrirlos mediante el intercambio de esta operación. Al final con ese resultado se vuelve a realizar una operación modular con el numero secreto y ambos sistemas obtienen la misma llave. Te anexo algunos links que me ayudaron a comprender un poco mejor este algoritmo y espero que no te haya enredado tanto con mi respuesta que es muy general. 
[https://www.computerweekly.com/es/definicion/Intercambio-de-claves-Diffie-Hellman-intercambio-de-claves-exponencial#:~:text=El%20intercambio%20de%20claves%20de,que%20la%20tarea%20de%20un ](url)
[https://www.youtube.com/watch?v=RtPm9ub2RlY&t=2s](url)

En cuanto a tu segunda pregunta, ambos algoritmos se realizan a nivel de bits, recuerda que una computadora no comprende nuestro lenguaje, sin embargo nos pareció mas practico para el caso de DES explicar su estructura y en que consistía cada parte del algoritmo a entrar en detalle con el resultado de cada operación, cosa que con AES no pudimos abordar de la misma manera porque en cada ronda se realizan un numero diferente de operaciones. Igualmente te anexo algunos links por si quieres entrar mas en profundidad con estos dos algoritmos o si quieres verlo implementado en código. 
[https://www.geeksforgeeks.org/data-encryption-standard-des-set-1/](url)
[https://www.tutorialspoint.com/cryptography/data_encryption_standard.htm](url)  
[https://www.tutorialspoint.com/cryptography/advanced_encryption_standard.htm](url)
